modules/core/src/shadertools/src/lib/inject-shader.js

```diff
@@ -21,10 +20,10 @@ export default function injectShader(source, type, inject, injectStandardStubs)
   for (const key in inject) {
     const fragment = inject[key];
     switch (key) {
-    // declarations are injected at beginning of shader
+      // declarations are injected before the main function
     case 'vs:#decl':
       if (isVertex) {
-        source = source.replace(REGEX_DECLARATIONS, match => match + fragment);
+        source = source.replace(REGEX_START_OF_MAIN, match => fragment + match);
```

Comment:
Insert a line break after fragment for safety.

Reply:
@Pessimistress it's updated